moto/iot/models.py

```diff
@@ -484,6 +484,74 @@ def to_dict(self):
         }
 
 
+class FakeDomainConfiguration(BaseModel):
+    def __init__(
+        self,
+        region_name,
+        domain_configuration_name,
+        domain_name,
+        server_certificate_arns,
+        domain_configuration_status,
+        service_type,
+        authorizer_config,
+        domain_type,
+    ):
+        if domain_configuration_status not in ["ENABLED", "DISABLED"]:
```

Comment:
This branch doesn't seem to be reachable, as we're always passing in a hardcoded `ENABLED` when creating an instance

Reply:
You're right; domain configurations seem to always start as ENABLED and the domain type can't be set by the user, AWS sets it automatically based on the domain, which is why I hardcoded them. I'll remove these two checks, they can be added in the future if these parameters are allowed to be set somehow.